Sources/NIOHTTP2/HTTP2PipelineHelpers.swift

```diff
@@ -422,6 +447,40 @@ extension ChannelPipeline.SynchronousOperations {
         // `multiplexer` will always be non-nil when we are initializing with an `inboundStreamInitializer`
         return try handler.syncMultiplexer()
     }
+
+    /// Synchronously configures a `ChannelPipeline` to speak HTTP/2.
+    ///
+    /// In general this is not entirely useful by itself, as HTTP/2 is a negotiated protocol. This helper does not handle negotiation.
+    /// Instead, this simply adds the handlers required to speak HTTP/2 after negotiation has completed, or when agreed by prior knowledge.
+    /// Whenever possible use this function to setup a HTTP/2 server pipeline, as it allows that pipeline to evolve without breaking your code.
+    ///
+    /// - Parameters:
+    ///   - mode: The mode this pipeline will operate in, server or client.
+    ///   - channel: to which the created``HTTP2StreamMultiplexer`` will belong.
+    ///   - initialLocalSettings: The settings that will be used when establishing the connection. These will be sent to the peer as part of the
+    ///         handshake.
+    ///   - position: The position in the pipeline into which to insert these handlers.
+    ///   - targetWindowSize: The target size of the HTTP/2 flow control window.
+    ///   - inboundStreamInitializer: A closure that will be called whenever the remote peer initiates a new stream. This should almost always
+    ///         be provided, especially on servers.
+    /// - Returns: An `EventLoopFuture` containing the `HTTP2StreamMultiplexer` inserted into this pipeline, which can be used to initiate new streams.
+    internal func configureHTTP2Pipeline(mode: NIOHTTP2Handler.ParserMode,
+                                         channel: Channel,
+                                         initialLocalSettings: [HTTP2Setting] = nioDefaultSettings,
+                                         position: ChannelPipeline.Position = .last,
+                                         targetWindowSize: Int = 65535,
+                                         inboundStreamInitializer: NIOChannelInitializer?) throws -> HTTP2StreamMultiplexer {
+        var handlers = [ChannelHandler]()
+        handlers.reserveCapacity(2)  // Update this if we need to add more handlers, to avoid unnecessary reallocation.
+
+        handlers.append(NIOHTTP2Handler(mode: mode, initialSettings: initialLocalSettings))
+        let multiplexer = HTTP2StreamMultiplexer(mode: mode, channel: channel, targetWindowSize: targetWindowSize, inboundStreamInitializer: inboundStreamInitializer)
+        handlers.append(multiplexer)
+
+        try self.addHandlers(handlers, position: position)
```

Comment:
Given we're doing synchronous operations here we can actually save an allocation by avoiding the `handlers` array and just calling `addHandler` multiple times.